src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java

```diff
@@ -1091,24 +1090,26 @@ private void getDocIdsVirtualServer(DocIdPusher pusher)
         }
         Map<GroupPrincipal, Collection<Principal>> siteDefs
             = siteAdaptor.computeMembersForGroups(site.getGroups());
-        for (Map.Entry<GroupPrincipal, Collection<Principal>> me
-            : siteDefs.entrySet()) {
-          defs.put(me.getKey(), me.getValue());
-          if (defs.size() >= feedMaxUrls) {
-            pusher.pushGroupDefinitions(defs, false);
-            defs.clear();
-          }
-        }
+        String siteId = site.getMetadata().getID();
+        sitePushGroupDefinitions(siteId, pusher, siteDefs);
       }
       if (excluded.size() > 0) {
         log.log(Level.INFO,
             "List of site collections excluded from index in "
                 + "getDocIds: {0}", excluded);
       }
-    }
-    pusher.pushGroupDefinitions(defs, false);
+    }    
     log.exiting("SharePointAdaptor", "getDocIdsVirtualServer");
   }
+  
+  private void sitePushGroupDefinitions(String siteID, DocIdPusher pusher,
+      Map<GroupPrincipal, Collection<Principal>> groupDefs)
+      throws InterruptedException {
+    String sourceID = siteID.replaceAll("[{}]", "");
+    sourceID = "SITEID-"+ sourceID;
+    pusher.pushGroupDefinitions(groupDefs, false, DocIdPusher.FeedType.REPLACE,
```

Comment:
Please replace 'false' with DocIdPusher.EVERYTHING_CASE_INSENSITIVE. I would also recommend statically importing DocIdPusher.EVERYTHING_CASE_INSENSITIVE and DocIdPusher.FeedType.

Reply:
Done